packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -312,8 +329,14 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 
 		const inverses: TaggedChange<TChange>[] = [];
 		for (let i = commits.length - 1; i >= 0; i--) {
-			const inverse = this.changeFamily.rebaser.invert(commits[i], false);
-			inverses.push(tagRollbackInverse(inverse, this.mintRevisionTag(), commits[i].revision));
+			const revision = this.mintRevisionTag();
+			const inverse = this.changeFamily.rebaser.replaceRevisions(
+				this.changeFamily.rebaser.invert(commits[i], false),
```

Comment:
Would it be better to have invert take a RevisionTag instead? (possibly in a follow-up PR)